src/v/storage/index_state.h

```diff
@@ -89,11 +89,13 @@ class offset_time_index {
    1 byte  - batch_timestamps_are_monotonic
    1 byte  - with_offset
    1 byte  - non_data_timestamps
+   8 bytes - first_data_offset
  */
 struct index_state
-  : serde::envelope<index_state, serde::version<6>, serde::compat_version<4>> {
+  : serde::envelope<index_state, serde::version<7>, serde::compat_version<4>> {
     static constexpr auto monotonic_timestamps_version = 5;
     static constexpr auto broker_timestamp_version = 6;
+    static constexpr auto first_data_offset_version = 7;
```

Comment:
i think you need to update read_nested and serde_write in index_state.cc

Reply:
Done